plugins/webapi/value/sendtransaction/handler.go

```diff
@@ -22,6 +22,19 @@ func Handler(c echo.Context) error {
 		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
 	}
 
+	_, cachedInputs, consumedBalances, _, err := valuetransfers.Tangle.RetrieveConsumedInputDetails(tx)
+	defer cachedInputs.Release()
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, Response{Error: err.Error()})
+	}
+	if !valuetransfers.Tangle.CheckTransactionOutputs(consumedBalances, tx.Outputs()) {
+		return c.JSON(http.StatusBadRequest, Response{Error: "Transaction does not spend all funds from inputs"})
+	}
+
+	if !tx.SignaturesValid() {
+		return c.JSON(http.StatusBadRequest, Response{Error: "Invalid transaction signatures"})
+	}
+
```

Comment:
Move this validation logic into a function within the value layer package since with high certainty it will be needed somewhere else too.